pkgs/build-support/emacs/melpa.nix

```diff
@@ -35,25 +35,31 @@ in
   pname
   /*
     ename: Original Emacs package name, possibly containing special symbols.
+    Default: pname
   */
-, ename ? null
+, ename ? pname
 , version
-, recipe
+  /*
+    commit: Optioal package history commit.  Default: src.rev or "unknown"
+    This will be written into the generated package but it not needed during
+    the build process.
+  */
+, commit ? (args.src.rev or "unknown")
+  /*
+    recipe: Optional MELPA recipe.  Default: a minimal but not accurate recipe
+    You need to provide your own recipe if you want to set :file.
+  */
+, recipe ? (writeText "${pname}-recipe" ''(${ename} :fetcher git :url "")'')
```

Comment:
Well, I understand this is not accurate, but there are other fetchers too. 
The recipe description is a bit free-form:

https://github.com/melpa/melpa?tab=readme-ov-file#recipe-format

(I am cogitating something like a `writeMelpaRecipe` function that converts an input set to the format above.)

Reply:
I think the current implementation is good enough and reaches a good balance between complexity and functionality.

After reading related code of `melpaBuild`, [`package-recipe`][2] and [`package-build`][3], I find that only `:name`, `:version`, `:commit` and `:files` can affect the build process of `melpaBuild`.  `:fetcher` and `:url` (or `:repo`) are merely used to pass the [package-recipe--validate][1].  They cannot affect the build process of `melpaBuild`.  So I do not think extra complexity to provide a more accurate recipe is justified.

[1]: https://github.com/melpa/package-build/blob/d9ce37b82529827ce720ad3ccf4af887d19fcfb0/package-recipe.el#L162
[2]: https://github.com/melpa/package-build/blob/master/package-recipe.el
[3]: https://github.com/melpa/package-build/blob/master/package-build.el

Reply:
> I am cogitating something like a `writeMelpaRecipe` function that converts an input set to the format above.

This is useful if users want to provide their own recipe instead of using the default value.

I think the only case they want to do so is to set `:files`.  To improve this UX of this use case, my proposal is to add another argument `files ? null` for `melpaBuild` and use it for the default value of `recipe`.

```nix
, files ? null
, recipe ? (writeText "${pname}-recipe" ''
    (${ename} :fetcher git :url ""
              ${if files != null then ":files " + files else ""})
  '')
```

Reply:
Well, given that it will not be worse than the current situation, then fine.

Reply:
> To improve UX of this use case, my proposal is to add another argument `files ? null` for `melpaBuild` and use it for the default value of `recipe`.

Implemented.